code/modules/mob/living/silicon/robot/inventory.dm

```diff
@@ -290,9 +290,14 @@
 		return ..(W)
 
 
-/mob/living/silicon/robot/canUnEquip(obj/item/I) //Force overrides NODROP for things like wizarditis and admin undress.
-	if(!I || !I.loc)
-		return TRUE
-	if (istype(I.loc, /obj/item/weapon/gripper)) //Robots are allowed to drop the things in their gripper
-		return TRUE
-	return ..(I) //This will be false for things directly equipped
\ No newline at end of file
+/mob/living/silicon/robot/can_unequip(obj/item/I)
+	//Robots are allowed to drop the things in their gripper
+	if(I && istype(module_active, /obj/item/weapon/gripper))
+		var/obj/item/weapon/gripper/G = module_active
+		return I == G || G.wrapped == I
```

Comment:
I == G ?
That would allow the robot to drop the gripper itself. That is bad.

Also I used checking loc of gripper to account for possible buggy situations where something would cease to be in the gripped variable but still be held inside the gripper